File: aasexplorer-core/src/main/java/com/ficture7/aasexplorer/model/MarkingScheme.java
package com.ficture7.aasexplorer.model;

import com.ficture7.aasexplorer.util.ObjectUtil;

import static com.ficture7.aasexplorer.util.ObjectUtil.checkNotNull;

/**
 * Represents a marking scheme resource.
 *
 * @author FICTURE7
 */
public class MarkingScheme extends Resource {

    private final Session session;

    /**
     * Constructs a new instance of the {@link MarkingScheme} class with the specified resource name
     * and session.
     *
     * @param name    Resource name.
     * @param session Session.
     * @throws NullPointerException {@code name} is null.
     * @throws NullPointerException {@code session} is null.
     */
    public MarkingScheme(String name, Session session) {
        super(name);

        this.session = checkNotNull(session, "session");
    }

    /**
     * Gets the {@link Session} of the {@link MarkingScheme}.
     *
     * @return {@link Session} of the {@link MarkingScheme}.
     */
    public Session session() {
        return session;
    }
}

File: aasexplorer-core/src/main/java/com/ficture7/aasexplorer/model/MarkingScheme.kt
package com.ficture7.aasexplorer.model

/**
 * Represents a marking scheme resource.
 *
 * @author FICTURE7
 */
class MarkingScheme : Resource {

    /**
     * Constructs a new instance of the [MarkingScheme] class with the specified resource name
     * and session.
     *
     * @param name    Resource name.
     * @param session Session.
     */
    constructor(name: String, session: Session) : super(name) {
        this.session = session
    }

    /**
     * Returns the [Session] of the [MarkingScheme].
     *
     * @return [Session] of the [MarkingScheme].
     */
    val session: Session
}